File: prodplus-api/src/main/java/br/com/prodplus/models/EstoqueProduto.java
package br.com.prodplus.models;

import java.time.LocalDateTime;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.validation.constraints.NotNull;

import br.com.prodplus.models.enums.TipoEntrada;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 
 * @author Marlon Fernando Garcia
 *
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class EstoqueProduto extends Estoque {

	private static final long serialVersionUID = -2164031057795902091L;
	@ManyToOne
	@JoinColumn(nullable = false)
	@NotNull(message = "o produto é obrigatório!")
	private Produto produto;

	@Builder
	public EstoqueProduto(Integer id, LocalDateTime data, Double quantidade, Double entrada,
			TipoEntrada tipo, Produto produto) {
		super(id, data, quantidade, entrada, tipo);
		this.setProduto(produto);
	}

}

File: prodplus/src/app/models/estoque-produto.ts
import { Produto } from './produto';

export class EstoqueProduto {
  id: number | null;
  data: string;
  quantidade: number;
  entrada: number;
  tipo: string;
  produto: Produto;

  constructor(
    data: string,
    quantidade: number,
    entrada: number,
    tipo: string,
    produto: Produto,
    id?: number
  ) {
    this.data = data;
    this.quantidade = quantidade;
    this.entrada = entrada;
    this.tipo = tipo;
    this.produto = produto;
    if (typeof id === 'undefined') this.id = null;
    else this.id = id;
  }
}
